OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -451,8 +449,8 @@ public bool AircraftCanEnter(Actor a)
 			if (self.AppearsHostileTo(a))
 				return false;
 
-			return Info.RearmBuildings.Contains(a.Info.Name)
-				|| Info.RepairBuildings.Contains(a.Info.Name);
+			return (rearmableInfo != null && rearmableInfo.RearmActors.Contains(a.Info.Name))
```

Comment:
Hard to say for sure (I'm not *that* far ahead locally), but if we can get the PRs reviewed & merged somewhat consistently, it might not be that far off.